presto-hive/src/test/java/io/prestosql/plugin/hive/parquet/ParquetDecimalScalingTest.java

```diff
@@ -0,0 +1,328 @@
+        MessageType schema = parseMessageType(format("message record { %s; }", Joiner.on("; ").join(fields)));
+        List<ObjectInspector> inspectors = inserts.stream().map(ParquetDecimalInsert::getParquetObjectInspector).collect(toImmutableList());
+        List<String> columnNames = inserts.stream().map(ParquetDecimalInsert::getColumnName).collect(toImmutableList());
+        Iterator<?>[] values = inserts.stream().map(ParquetDecimalInsert::getValues).map(Iterable::iterator).toArray(Iterator[]::new);
+
+        createParquetFile(
+                output,
+                getStandardStructObjectInspector(columnNames, inspectors),
+                values,
+                schema,
+                Collections.singletonList("record"));
+    }
+
+    private static Properties createTableProperties(List<String> columnNames, List<ObjectInspector> objectInspectors)
+    {
+        Properties orderTableProperties = new Properties();
+        orderTableProperties.setProperty("columns", Joiner.on(',').join(columnNames));
+        orderTableProperties.setProperty("columns.types", Joiner.on(',').join(transform(objectInspectors, ObjectInspector::getTypeName)));
+        return orderTableProperties;
+    }
+
+    public static class ParquetDecimalInsert
+    {
+        private final String columnName;
+        private final boolean forceFixedLengthArray;
+        private final int precision;
+        private final int scale;
+        private final List<String> values;
+
+        public ParquetDecimalInsert(String columnName, boolean forceFixedLengthArray, int precision, int scale, List<String> values)
+        {
+            this.columnName = columnName;
+            this.forceFixedLengthArray = forceFixedLengthArray;
+            this.precision = precision;
+            this.scale = scale;
+            this.values = values;
+        }
+
+        public String getColumnName()
+        {
+            return columnName;
+        }
+
+        public String parquetStorage()
+        {
+            if (!forceFixedLengthArray && precision > 0 && precision < 10) {
+                return "INT32";
+            }
+
+            if (!forceFixedLengthArray && precision >= 10 && precision < 18) {
+                return "INT64";
+            }
+
+            if (precision > 38 || precision < 0) {
+                throw new IllegalArgumentException("Scale could not be greater than 38 or less than 0");
+            }
+
+            return format("fixed_len_byte_array(%d)", ParquetHiveSerDe.PRECISION_TO_BYTE_COUNT[precision - 1]);
+        }
+
+        public ObjectInspector getParquetObjectInspector()
+        {
+            if (!forceFixedLengthArray && precision > 0 && precision < 10) {
+                return javaIntObjectInspector;
+            }
+
+            if (!forceFixedLengthArray && precision >= 10 && precision < 18) {
+                return javaLongObjectInspector;
```

Comment:
is it suitable also for scale!=0 ?

Reply:
yes, it's how the decimal is represented in parquet (decimals got converted to longs)